plugin/storage/es/spanstore/reader.go

```diff
@@ -139,24 +140,44 @@ type timeRangeIndexFn func(indexName string, indexDateLayout string, startTime t
 
 type sourceFn func(query elastic.Query, nextTime uint64) *elastic.SearchSource
 
-func getTimeRangeIndexFn(archive, useReadWriteAliases bool) timeRangeIndexFn {
+func getTimeRangeIndexFn(archive, useReadWriteAliases bool, remoteReadClusters []string) timeRangeIndexFn {
 	if archive {
 		var archiveSuffix string
 		if useReadWriteAliases {
 			archiveSuffix = archiveReadIndexSuffix
 		} else {
 			archiveSuffix = archiveIndexSuffix
 		}
-		return func(indexName, indexDateLayout string, startTime time.Time, endTime time.Time) []string {
-			return []string{archiveIndex(indexName, archiveSuffix)}
-		}
+		return addRemoteReadClusters(func(indexPrefix, indexDateLayout string, startTime time.Time, endTime time.Time) []string {
+			return []string{archiveIndex(indexPrefix, archiveSuffix)}
+		}, remoteReadClusters)
 	}
 	if useReadWriteAliases {
-		return func(indices string, indexDateLayout string, startTime time.Time, endTime time.Time) []string {
-			return []string{indices + "read"}
+		return addRemoteReadClusters(func(indexPrefix string, indexDateLayout string, startTime time.Time, endTime time.Time) []string {
+			return []string{indexPrefix + "read"}
+		}, remoteReadClusters)
+	}
+	return addRemoteReadClusters(timeRangeIndices, remoteReadClusters)
+}
+
+// Add a remote cluster prefix for each cluster and for each index and add it to the list of original indices.
+// Elasticsearch cross cluster api example GET /twitter,cluster_one:twitter,cluster_two:twitter/_search.
+func addRemoteReadClusters(fn timeRangeIndexFn, remoteReadClusters []string) timeRangeIndexFn {
+	return func(indexPrefix string, indexDateLayout string, startTime time.Time, endTime time.Time) []string {
+		jaegerIndices := fn(indexPrefix, indexDateLayout, startTime, endTime)
+		if len(remoteReadClusters) == 0 {
+			return jaegerIndices
 		}
+
+		for _, jaegerIndex := range jaegerIndices {
+			for _, remoteCluster := range remoteReadClusters {
```

Comment:
Related to the earlier comment, if `remoteReadClusters` defaults to `[]string{""}`, the returned indices count would be doubled with the new indices looking like `:<index-name>`. Is that expected?

Reply:
This should be avoided now with the changes to `plugin/storage/es/options.go`